### PR TITLE
handling data in signup and signin form using hooks

### PR DESCRIPTION
**Pull request by afif1400**
Check if the implementation is correct, and want me to continue.

### CODE REVIEW

File: src/auth/components/SignUpForm.js [hunk 1]
```diff
@@ -1,98 +1,90 @@
...
 import { Form, Icon, Input } from 'antd';
 import { FormButton } from './FormButton';
 import { ErrorMessage } from './ErrorMessage';
 
-const FormItem = Form.Item;
+const SignUpForm = ({ form, onSubmit }) => {
+    const [username, setUsername] = useState('');
+    const [email, setEmail] = useState('');
+    const [passwordOne, setPasswordOne] = useState('');
+    // eslint-disable-next-line
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:12]: Why do we need this disabling?

File: src/auth/components/SignUpForm.js [hunk 2]
```diff
@@ -1,98 +1,90 @@
...
+                        <Input
+                            prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
+                            type="password"
+                            placeholder="Password"
+                            onChange={event => setPasswordOne(event.target.value)}
+                        />
+                    )}
+                </FormItem>
+                <FormItem>
+                    {getFieldDecorator('passwordTwo', {
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:67]: Lets rename 'passwordTwo' to 'confrimPassword'
And 'password' to 'newPassword'

File: src/auth/components/SignUpForm.js [hunk 3]
```diff
@@ -1,98 +1,90 @@
...
+                        <Input
+                            prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
+                            type="password"
+                            placeholder="Confirm password"
+                            onChange={event => setPasswordTwo(event.target.value)}
+                        />
+                    )}
+                </FormItem>
+                <FormItem>
+                    <FormButton type="primary" htmlType="submit" className="login-form-button">
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:79]: It also important to make the button disabled while we making request, can you add this please here?

File: src/auth/components/SignUpForm.js [hunk 4]
```diff
@@ -1,98 +1,90 @@
...
+                            prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
+                            type="password"
+                            placeholder="Password"
+                            onChange={event => setPasswordOne(event.target.value)}
+                        />
+                    )}
+                </FormItem>
+                <FormItem>
+                    {getFieldDecorator('passwordTwo', {
+                        rules: [{ required: true, message: 'Please input your passwordTwo!' }],
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:68]: 'Please input your confirm password'

File: src/auth/components/SignUpForm.js [hunk 5]
```diff
@@ -1,98 +1,90 @@
...
+                        <Input
+                            prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
+                            placeholder="Email"
+                            onChange={event => setEmail(event.target.value)}
+                        />
+                    )}
+                </FormItem>
+                <FormItem>
+                    {getFieldDecorator('passwordOne', {
+                        rules: [{ required: true, message: 'Please input your passwordOne!' }],
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:56]: 'Please input your new password'

File: src/auth/components/SignUpForm.js [hunk 6]
```diff
@@ -1,98 +1,90 @@
-import React, { Component } from 'react';
-import { byPropKey } from '../../utils';
+import React, { useState } from 'react';
+// import { byPropKey } from '../../utils';
```
Comment (bmarvinb) [inline src/auth/components/SignUpForm.js:2]: remove commented code

**Review by bmarvinb — COMMENTED** [review]
Write some small comments, but in general everything is good

File: src/auth/components/SignUpForm.js (hunk 2 above)
Comment (afif1400) [reply src/auth/components/SignUpForm.js:67]: Sure

File: src/auth/components/SignUpForm.js (hunk 3 above)
Comment (afif1400) [reply src/auth/components/SignUpForm.js:79]: Okay

File: src/auth/components/SignUpForm.js (hunk 4 above)
Comment (afif1400) [reply src/auth/components/SignUpForm.js:68]: Will do.

File: src/auth/components/SignUpForm.js (hunk 6 above)
Comment (afif1400) [reply src/auth/components/SignUpForm.js:2]: Okay

File: src/auth/components/SignUpForm.js (hunk 1 above)
Comment (afif1400) [reply src/auth/components/SignUpForm.js:12]: Passwordtwo wasn't being used anywhere

Comment (bmarvinb) [reply src/auth/components/SignUpForm.js:12]: In production usually, we use the validator, that check if the new password and confirm password are correct, I will create new issue for this, or you can implement it if you want

Comment (afif1400) [reply src/auth/components/SignUpForm.js:12]: Yeah... I'll implement that np.

**Comment by afif1400** [comment]
#31 made all the changes as asked in conversation.

**Comment by bmarvinb** [comment]
@afif1400 Do you planning to add something else to this PR, or I can merge it?

**Comment by afif1400** [comment]
> @afif1400 Do you planning to add something else to this PR, or I can merge it?

You can merge I have also resolved issue #32

**Pull request merged**